addons/survey/controllers/main.py

```diff
@@ -732,35 +727,130 @@ def _get_user_input_domain(self, survey, line_filter_domain, **post):
+
+        # Get the matching user input lines
+        user_inputs_query = request.env['survey.user_input'].sudo()._search(user_input_domain)
+        user_input_lines = request.env['survey.user_input.line'].search([('user_input_id', 'in', user_inputs_query)])
+
+        return user_input_lines
+
+    @staticmethod
+    def _get_filters_from_post(post):
+        """ Extract the filters from post depending on the model that needs to be called to retrieve the filtered answer data.
+
+        * Model = 'A' : Match a `survey.question.answer` record (simple_choice, multiple_choice, matrix)
+        * Model = 'L' : Match a `survey.user_input.line` record (char_box, text_box, numerical_box, date, datetime)
+        :rtype: (collections.defaultdict[int, list[int]], list[int])
+        """
+        answer_cols_to_rows_mapping = defaultdict(list)
+        answer_user_input_line_ids = []
+
+        for data in post.get('filters', '').split('|'):
+            if not data:
+                break
+            model, row_id, ans_id = data.split(',')
+            row_id, ans_id = int(row_id), int(ans_id)
+            if model == 'A':
+                answer_cols_to_rows_mapping[ans_id].append(row_id)
+            elif model == 'L' and not row_id:
+                answer_user_input_line_ids.append(ans_id)
+
+        return answer_cols_to_rows_mapping, answer_user_input_line_ids
+
+    @staticmethod
+    def _get_value_with_operator(record, row_id=0):
+        """ Get the answer's value and domain operator to find answers with the
+        same value (depends on answer's model and question type).
+
+        :param record: 'survey.question.answer' or 'survey.user_input.line' record
+        :rtype: (string, string)
+        """
+        if record.question_id.question_type in ('simple_choice', 'multiple_choice', 'matrix'):
+            return f'{f"{row_id.value} : " if row_id else ""}{record.value}', None
+        return record[f'value_{record.question_id.question_type}'], {
+            'text_box': 'ilike',
+            'char_box': '=ilike',
+            'date': '=',
+            'datetime': '=',
+            'numerical_box': '=',
+        }[record.question_id.question_type]
+
+    @staticmethod
+    def _prepare_matching_answer_domain(question_type, answer, row_id=0, operator=None, value=None):
+        if question_type == "matrix":
+            return [expression.AND([[('question_id', '=', answer.question_id.id)], [('matrix_row_id', '=', row_id)],
+                                   [('suggested_answer_id', '=', answer.id)]])]
+        elif question_type in ('multiple_choice', 'simple_choice'):
+            return [expression.AND([[('question_id', '=', answer.question_id.id)], [('suggested_answer_id', '=', answer.id)]])]
+        elif question_type in ('char_box', 'text_box', 'numerical_box', 'date', 'datetime'):
+            return [expression.AND([[('question_id', '=', answer.question_id.id)], [(f'value_{answer.answer_type}', operator, value)]])]
```

Comment:
We tend to avoid using these kind of constructs: `f'value_{answer.answer_type}'`.

Although at first glance it seems handy and reduces the amount of code, it can in fact become an issue since it will prevent one from greping "value_char_box" and seeing where it's used (if let's say we somehow want to rename it).